tests/experimental/pydantic/test_basic.py

```diff
@@ -63,6 +64,73 @@ class UserType:
     assert field2.type.of_type is str
 
 
+def test_basic_type_with_input_metadata():
+    class User1(pydantic.BaseModel):
+        friend: Optional[str] = "friend_value"
+        frenemy: str
+
+    @strawberry.experimental.pydantic.type(User1, with_input_metadata=True)
+    class UserType1:
+        friend: strawberry.auto
+        frenemy: strawberry.auto
+
+    user_input_1 = UserType1(frenemy="frenemy_value")
+    assert is_unset(user_input_1.friend)
+    assert user_input_1.to_pydantic().friend == "friend_value"
+    assert user_input_1.to_pydantic().dict() == {
+        "friend": "friend_value",
+        "frenemy": "frenemy_value",
+    }
+    assert user_input_1.to_pydantic().dict(exclude_unset=True) == {
+        "frenemy": "frenemy_value"
+    }
+
+    class User2(pydantic.BaseModel):
+        friend: Optional[str] = pydantic.Field(default_factory=lambda: "friend_value")
+        frenemy: str
+
+    @strawberry.experimental.pydantic.type(User2, with_input_metadata=True)
+    class UserType2:
+        friend: strawberry.auto
+        frenemy: strawberry.auto
+
+    user_input_2 = UserType2(frenemy="frenemy_value")
+    assert is_unset(user_input_2.friend)
+    assert user_input_2.to_pydantic().friend == "friend_value"
+    assert user_input_2.to_pydantic().dict() == {
+        "friend": "friend_value",
+        "frenemy": "frenemy_value",
+    }
+    assert user_input_2.to_pydantic().dict(exclude_unset=True) == {
+        "frenemy": "frenemy_value"
+    }
+
```

Comment:
hmmm if you try this you can see that if you pass the default explictly, the value from the passed value doesn't get used

```
    # test when we pass the default value
    user_input_default = UserType2(
        frenemy="frenemy_value", friend="not_default_friend"
    )
    assert user_input_default.to_pydantic().friend == "not_default_friend"
    assert user_input_2.to_pydantic().dict() == {
        "friend": "not_default_friend",
        "frenemy": "frenemy_value",
    }
    assert user_input_2.to_pydantic().dict(exclude_unset=True) == {
        "friend": "not_default_friend",
        "frenemy": "frenemy_value",
    }
```

Reply:
Hey @thejaminator  - Just want to make sure here - you are creating an input called user_input_default and then testing user_input_2 (which is the one created at line 97). I believe there may have been a copy paste mix up? Hit me up on discord if we need to discuss something!

Reply:
sure, i replied to you what i want to test for in the discord chat https://discord.com/channels/689806334337482765/928742670447304734/930456003915444224